### PR TITLE
feat(proto): split proto files based on service

### PR DESCRIPTION
**Pull request by akhilerm**
split proto files for bdev/json rpc service

Signed-off-by: Akhil Mohan <akhil.mohan@mayadata.io>

### CODE REVIEW

File: protobuf/v1/mayastor.proto [hunk 1]
```diff
@@ -7,68 +7,37 @@ syntax = "proto3";
...
   // Call a (SPDK) json-rpc method
-  rpc JsonRpcCall (JsonRpcRequest) returns (JsonRpcReply) {}
+  rpc JsonRpcCall (JsonRpcRequest) returns (JsonRpcResponse) {}
 }
 
-message JsonRpcRequest {
-  string method = 1;
-  string params = 2;
+message Null {
+  google.protobuf.NullValue null = 1;
```
Comment (akhilerm) [inline protobuf/v1/mayastor.proto:35]: @gila This is the Null message you wanted on `protobuf.NullValue` right?

**Comment by akhilerm** [comment]
cc: @shubham14bajpai Can you take a look on this PR?

**Review by shubham14bajpai — APPROVED** [review]

File: protobuf/v1/bdev_json.proto [hunk 2]
```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package mayastor.v1;
```
Comment (gila) [inline protobuf/v1/bdev_json.proto:3]: Why is the file called bdev_json.proto? that is confusing?

File: protobuf/v1/mayastor.proto (hunk 1 above)
Comment (gila) [reply protobuf/v1/mayastor.proto:35]: well look here:

``` rpc Unshare(UnshareRequest) returns (Null) {}```

Unshare returns "nothing" it says... So can we simply do returns() ? Some goes for Destroy etc.

File: protobuf/v1/mayastor.proto [hunk 3]
```diff
@@ -7,68 +7,37 @@ syntax = "proto3";
...
-}
-
-message BdevUri {
-  string uri = 1;
-}
-
-message CreateReply {
-  string name = 1;
+  // list the block devices
+  rpc List(NullableString) returns (ListBdevResponse) {}
```
Comment (gila) [inline protobuf/v1/mayastor.proto:17]: Why do dont have ` v1/bdev.proto` and have all the services defined in there?

File: protobuf/v1/mayastor.proto (hunk 1 above)
Comment (akhilerm) [reply protobuf/v1/mayastor.proto:35]: >  So can we simply do returns() 

This is not the right way in grpc. The proto file wont compile. It expects a return message. 
```
rpc Unshare(UnshareRequest) returns () {}
```

we could use the `google.protobuf.Empty` and write like 
```
rpc Unshare(UnshareRequest) returns (google.protobuf.Empty) {}
```

**Pull request merged**